### PR TITLE
adding a new logging doc

### PR DESCRIPTION
**Pull request by robertpatrick**

### CODE REVIEW

**Review by rakillen — APPROVED** [review]

**Review by CarolynRountree — APPROVED** [review]

File: documentation/2.0/content/userguide/tools-config/logging.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+---
+title: "Logging"
+date: 2022-02-04T17:18:00-05:00
+draft: false
+weight: 4
+---
+
+The WebLogic Deploy Tooling bas a built-in logging framework based on `java.util.logging`.  Its logging configuration
```
Comment (rosemarymarano) [inline documentation/2.0/content/userguide/tools-config/logging.md:8]: bas (typo) -> has

File: documentation/2.0/content/userguide/tools-config/logging.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+is specified in `$WDT_HOME/etc/logging.properties`.  By default, the logging framework writes to both the console and
+a log file.
+
+#### Log File
```
Comment (rosemarymarano) [inline documentation/2.0/content/userguide/tools-config/logging.md:12]: Log File -> Log file (internal headings get sentence capitalization versus title capitalization; supposedly a more "conversational/less formal" presentation)

File: documentation/2.0/content/userguide/tools-config/logging.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+By default, WDT tools write their log files to the `$WDT_HOME/logs` directory and the log file name reflects the name of the tool.
+For example, if you run the `validateModel` tool then the log file will be `$WDT_HOME/logs/validateModel.log`.  These
+log files are overwritten each time you run a particular tool so the file only contains the logs from the last tool
```
Comment (rosemarymarano) [inline documentation/2.0/content/userguide/tools-config/logging.md:15]: only contains -> contains only (Position of only: Keep only close to that which it modifies to improve the clarity and accuracy of the sentence.)

File: documentation/2.0/content/userguide/tools-config/logging.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+a location to write the logs.  The user must have write permission on the directory in order for it to be selected.
+The search order is as follows:
+
+- Check the `WLSDEPLOY_LOG_DIRECTORY` environment variable.
+- Check the current working directory (as defined by the `user.dir` Java system property) and create a `logs` subdirectory.
+- Check the temp directory (as defined by the `java.io.tmpdir` Java system property) and create a `wdt-logs` subdirectory.
+
+If none of these locations are writable, the logging framework prints an error message to `stderr` and exits.
+
+#### Console Output
```
Comment (rosemarymarano) [inline documentation/2.0/content/userguide/tools-config/logging.md:28]: Console Output -> Console output

File: documentation/2.0/content/userguide/tools-config/logging.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+WDT tools output logging information to `stdout` and `stderr`, as appropriate.  By default, only `INFO` level messages
+are sent to `stdout`.  All `WARNING` and `SEVERE` messages are set to `stderr`.  In addition to normal log messages
```
Comment (rosemarymarano) [inline documentation/2.0/content/userguide/tools-config/logging.md:30]: normal log messages -> typical log messages

File: documentation/2.0/content/userguide/tools-config/logging.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+overview of the tool execution status.  For example, the `validateModel` tool execution with no warnings or errors will
+produce output that looks similar to this:
+
+```
+Issue Log for validateModel version 2.0.0 running WebLogic version 12.2.1.4.0.210930 offline mode:
+
+Total:       WARNING :     0    SEVERE :     0
+```
+
+#### Logging Levels
```
Comment (rosemarymarano) [inline documentation/2.0/content/userguide/tools-config/logging.md:41]: Logging levels

File: documentation/2.0/content/userguide/tools-config/logging.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+As mentioned previously, WDT's logging framework is based on `java.util.logging` so all logging levels defined in
+the `java.utiul.logging.Level` class apply to WDT loggers.  For a quick review of those levels, see the
+[javadoc](https://docs.oracle.com/javase/8/docs/api/java/util/logging/Level.html).
+
+WDT uses hierarchical loggers that align with what parts of the code is executing.  The root logger is named `wlsdeploy`.
```
Comment (rosemarymarano) [inline documentation/2.0/content/userguide/tools-config/logging.md:46]: align with what parts of the code is executing -> (reads awkwardly) -> align with the executing parts of the code.

File: documentation/2.0/content/userguide/tools-config/logging.md [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+Many loggers exist underneath the root logger; for example, `wlsdeploy.create`, `wlsdeploy.discover`, and `wlsdeploy.util`.
+By default, the WDT `logging.properties` file sets the logging level of the root and several important loggers.  If
+level for a particular logger is not set, that logger will use the level of its parent logger.  This delegation to the
```
Comment (rosemarymarano) [inline documentation/2.0/content/userguide/tools-config/logging.md:49]: If level -> If the level

File: documentation/2.0/content/userguide/tools-config/logging.md [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+`wlsdeploy.debugToStdout` Java system property to allow debug log messages (those logged at the `FINE` level or below)
+to appear in `stdout` as long as the loggers to which the messages are logged are not filtering those log levels.  For
+example, doing the following will cause debug output to be written to the console:
+
+```
+export WLSDEPLOY_PROPERTIES=-Dwlsdeploy.debugToStdout=true
+weblogic-deploy/bin/prepareModel.sh ...
+```
+
+#### Log Handlers
```
Comment (rosemarymarano) [inline documentation/2.0/content/userguide/tools-config/logging.md:64]: Log handlers

File: documentation/2.0/content/userguide/tools-config/logging.md [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+ | `java.util.logging.FileHandler.level`                | `OFF`                            | No logging output will be saved in the log file.                                                                     |
+ | `oracle.weblogic.deploy.logging.StdoutHandler.level` | `OFF`                            | No `INFO` level logging output will be written to the console.                                                       |
+ | `oracle.weblogic.deploy.logging.StderrHandler.level` | `OFF`                            | No `WARNING` or `ERROR` level logging output will be written to the console.                                         |
+ | `oracle.weblogic.deploy.logging.SummaryHandler.level` | `OFF`                            | No tool execution summary output will be written to the console.                                                     |
+ | `oracle.weblogic.deploy.logging.SummaryHandler.size` | Any number                       | Limits the number of memory-buffered `WARNING` and `ERROR` log records (default is 3000).                            |
+
+Use the `WLSDEPLOY_LOG_HANDLERS` environment variable ss an alternative to specifying the list of handlers in the
```
Comment (rosemarymarano) [inline documentation/2.0/content/userguide/tools-config/logging.md:86]: ss an ->  is an

**Review by rosemarymarano — COMMENTED** [review]
Minor edits

**Pull request merged**